File: 14.cpp
struct Solution {
    string longestCommonPrefix(const vector<string>& strs) {
        if (strs.size() == 1){
            return strs.front();
        }
        size_t ptr = strs[0].size();
        for (int i = 1; i < strs.size(); ++i){
            ptr = min(ptr, strs[i].size());
            for (int j = 0; j < ptr; ++j){
                if (strs[0][j] != strs[i][j]){
                    ptr = j;
                    break;
                }
            }
        }
        return strs[0].substr(0, ptr);
    }
};

File: sol/14/14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]
        ptr = len(strs[0])
        for i in range(1, len(strs)):
            ptr = min(ptr, len(strs[i]))
            for j in range(ptr):
                if strs[0][j] != strs[i][j]:
                    ptr = j
                    break
        return strs[0][:ptr]
